core/src/tile/tileManager.cpp

```diff
@@ -173,8 +188,8 @@ void TileManager::updateTileSet() {
 
             } else if (curTileId == NOT_A_TILE || visTileId < curTileId) {
                 // tileSet is missing an element present in visibleTiles
-                addTile(visTileId);
-                enqueueLoadTask(visTileId, viewCenter);
+                if (!addTile(visTileId))
+                    enqueueLoadTask(visTileId, viewCenter);
```

Comment:
yea.. we got some more suspects

    grep -E -r --include \*.h --include \*.cpp "if ?\(" | grep -v {